ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -182,19 +260,29 @@ instance SameDepIndex (BlockQuery blk) => SameDepIndex (Query blk) where
     = Just Refl
   sameDepIndex GetSystemStart _
     = Nothing
+  sameDepIndex GetChainBlockNo GetChainBlockNo
+    = Just Refl
+  sameDepIndex GetChainBlockNo _
+    = Nothing
+  sameDepIndex GetChainPoint GetChainPoint
+    = Just Refl
+  sameDepIndex GetChainPoint _
+    = Nothing
 
 deriving instance Show (BlockQuery blk result) => Show (Query blk result)
 
 -- | Answer the given query about the extended ledger state.
-answerQuery ::
-     (QueryLedger blk, ConfigSupportsNode blk)
+answerQuery :: forall blk result.
+     (QueryLedger blk, ConfigSupportsNode blk, HasAnnTip blk)
   => ExtLedgerCfg blk
   -> Query blk result
   -> ExtLedgerState blk
   -> result
 answerQuery cfg query st = case query of
   BlockQuery blockQuery -> answerBlockQuery cfg blockQuery st
   GetSystemStart -> getSystemStart (topLevelConfigBlock (getExtLedgerCfg cfg))
+  GetChainBlockNo -> headerStateBlockNo (headerState st)
```

Comment:
About the names, can you remind me why did we chose `GetChainBlockNo` instead of `GetBlockNo` and why did we choose `GetChainPoint` instead of `GetPoint`? 

I have a similar question regarding the `headerState` prefix in the name of the functions used here.

Reply:
@dcoutts suggested the chain prefix which I presume means the block number of the tip of the chain as it is currently rather than any other block.  This has precedence in `cardano-api` we had a Shelley specific query called `QueryChainPoint` and this PR enables us to replace that with an era-independent query of the same name.

The `headerState` prefix I had was simply saying where I got the data from, so provenance rather than description.

Reply:
Ok, that makes sense. Thanks.